samples/application_development/number_crunching/README.rst

```diff
@@ -0,0 +1,130 @@
+.. _number_crunching:
+
+Number Crunching
+#####################
```

Comment:
I suggest using `:zephyr:code-sample:` directive (see guidelines in `doc/templates/sample.tmpl`) -- this might also help actually refine the scope/goal for this sample when completing the "This code sample shows how to ..." sentence